### PR TITLE
Add support for retrieving the LIP device list.

### PR DESCRIPTION
**Pull request by bdraco**
The LIP device list allows pico remotes to be identified
when connecting to devices that also support LIP (pro and ra select). Without
the lip device report the user has to guess which device
id is being reported.

This enabled pico remote support in Home Assistant for Pro and RA Select users via
https://github.com/home-assistant/core/pull/45315 https://github.com/home-assistant/core/commit/e546f6d122a65bb599016c81d8d0ab65f31c1716

### CODE REVIEW

File: tests/test_smartbridge.py [hunk 1]
```diff
@@ -180,25 +180,31 @@ async def _accept_connection(self, leap, wait):
         response.set_result(response_from_json_file("devices.json"))
         leap.requests.task_done()
 
-        # Second message should be read request on /virtualbutton
+        # First message should be read request on /server/2/id
```
Comment (mdonoughe) [inline tests/test_smartbridge.py:183]: Should this be "Second message"?

File: tests/test_smartbridge.py [hunk 2]
```diff
@@ -435,6 +441,138 @@ async def test_device_list(bridge: Bridge):
     assert devices[0]["device_id"] == "3"
 
 
+@pytest.mark.asyncio
+async def test_lip_device_list(bridge: Bridge):
+    """Test methods getting devices."""
+    devices = bridge.target.get_lip_devices()
+    expected_devices = {
```
Comment (mdonoughe) [inline tests/test_smartbridge.py:448]: Can we edit down the test data? It seems like one or two would be enough.

File: tests/test_smartbridge.py [hunk 3]
```diff
@@ -435,6 +441,138 @@ async def test_device_list(bridge: Bridge):
...
+        2: {
+            "Name": "Back Hall Pico",
+            "ID": 2,
+            "Area": {"Name": "Back Hall"},
+            "Buttons": [
+                {"Number": 2},
+                {"Number": 3},
+                {"Number": 4},
+                {"Number": 5},
+                {"Numbaer": 6},
```
Comment (mdonoughe) [inline tests/test_smartbridge.py:458]: This "a" is suspicious.

File: pylutron_caseta/smartbridge.py [hunk 4]
```diff
@@ -143,6 +144,10 @@ def get_devices(self) -> Dict[str, dict]:
         """Will return all known devices connected to the Smart Bridge."""
         return self.devices
 
+    def get_lip_devices(self) -> Dict[int, dict]:
```
Comment (mdonoughe) [inline pylutron_caseta/smartbridge.py:147]: If `lip_devices` is unprefixed, do we also need `get_lip_devices`?

Comment (bdraco) [reply pylutron_caseta/smartbridge.py:147]: Nope, I'll take it out.  It was only there to match `get_devices`

File: tests/test_smartbridge.py (hunk 3 above)
Comment (bdraco) [reply tests/test_smartbridge.py:458]: Copy paste error on my part. Will fix

File: tests/test_smartbridge.py (hunk 2 above)
Comment (bdraco) [reply tests/test_smartbridge.py:448]: I'll reduce it to two devices.

File: tests/test_smartbridge.py (hunk 1 above)
Comment (bdraco) [reply tests/test_smartbridge.py:183]: Yes. will fix

**Comment by bdraco** [comment]
TYVM. Would you mind bumping the pypi package as well?

**Comment by mdonoughe** [comment]
https://pypi.org/project/pylutron-caseta/0.9.0/

**Comment by bdraco** [comment]
> pypi.org/project/pylutron-caseta/0.9.0

TYVM!

**Pull request merged**